backends/qualcomm/builders/node_visitor.py

```diff
@@ -49,51 +51,37 @@ class NodeVisitor:
     Node visitor pattern for visiting nodes in an edge IR graph
     """
 
-    def __init__(self, external_ids, compile_mode) -> None:
+    def __init__(
+        self, external_ids, compile_mode, edge_program: torch.export.ExportedProgram
+    ) -> None:
         self.external_ids = external_ids or {}
         self.compile_mode = compile_mode
+        self.edge_program = edge_program
 
-    def is_graph_input(self, tensor: torch.fx.Node) -> bool:
-        """
-        Check if the given tensor is a graph input
-
-        Args:
-            tensor: EdgeIR Tensor that is being checked for graph input
-        """
-        return tensor.op == "placeholder"
-
-    def is_graph_output(self, tensor: torch.fx.Node) -> bool:
-        """
-        Check if the given tensor is used as a graph output
-
-        Args:
-            tensor: EdgeIR Tensor that is being checked for graph input
-        """
-
-        for user in tensor.users.keys():
-            if user.op == "output":
-                return True
-        return False
-
-    def get_tensor_shape(self, input_node, op_node, idx=None):
+    def get_tensor(self, input_node, op_node, idx=None):
         """
-        Get tensor shape with axis_order
+        Get tensor value/shape with axis_order
         """
 
-        def get_tensor(node, index):
+        def _get_tensor(node, index):
             if index is not None:
                 assert isinstance(index, int)
+                if is_parameter(node, self.edge_program):
```

Comment:
@angelayi @tugsbayasgalan can we introduce utils on the ExportedProgram to check if a node is parameter or buffer node and then have utils get_parameter?

Basically
`EdgeProragram.is_parameter(node)` `EdgeProgram.get_parameter(node)`

Reply:
I think we already have [these helper functions](https://github.com/pytorch/pytorch/blob/e39668770a9bb38379803845a13e3f0069db62ea/torch/_export/utils.py#L97), but maybe we can add to the exported program too.